src/absil/ilwrite.fs

```diff
@@ -4462,105 +4474,146 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
-        DumpDebugInfo outfile pdbData
+
+    let writeDumpDebugInfo pdbData dumpTo =
+        reportTime showTimes "Dump Debug Info"
+        let dump stream = DumpDebugInfo stream pdbData
+        match dumpTo with
+        | EmittedStream stream -> dump stream
+        | EmittedFile file ->
+            use stream = FileSystem.FileStreamCreateShim(file)
+            dump stream
+
 
     // Now we've done the bulk of the binary, do the PDB file and fixup the binary. 
-    begin match pdbfile with
-    | None -> ()
-    | Some fmdb when runningOnMono -> 
-        WriteMdbInfo fmdb outfile pdbData
+    let writePdb (pdbData, textV2P, debugDirectoryChunk, debugDataChunk) outfile pdbpath =
+        reportTime showTimes "Generate PDB Info"
+        // The ISymUnmanagedWriter2 require a file name. This is frequently used during PDB writing.
+        // Ensure a name is provided here in the case we were given only a Stream value.
+        let idd = WritePdbInfo fixupOverlappingSequencePoints showTimes outfile pdbpath pdbData
             
-    | Some fpdb -> 
+        // Now we have the debug data we can go back and fill in the debug directory in the image 
+        let fs2 = new FileStream(outfile, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Read, 0x1000, false)
+        use os2 = new BinaryWriter(fs2)
         try 
-            let idd = WritePdbInfo fixupOverlappingSequencePoints showTimes outfile fpdb pdbData
-            reportTime showTimes "Generate PDB Info"
-            
-          // Now we have the debug data we can go back and fill in the debug directory in the image 
-            let fs2 = new FileStream(outfile, FileMode.OpenOrCreate, FileAccess.Write, FileShare.Read, 0x1000, false)
-            let os2 = new BinaryWriter(fs2)
-            try 
-                // write the IMAGE_DEBUG_DIRECTORY 
-                os2.BaseStream.Seek (int64 (textV2P debugDirectoryChunk.addr), SeekOrigin.Begin) |> ignore
-                writeInt32 os2 idd.iddCharacteristics // IMAGE_DEBUG_DIRECTORY.Characteristics 
-                writeInt32 os2 timestamp
-                writeInt32AsUInt16 os2 idd.iddMajorVersion
-                writeInt32AsUInt16 os2 idd.iddMinorVersion
-                writeInt32 os2 idd.iddType
-                writeInt32 os2 idd.iddData.Length  // IMAGE_DEBUG_DIRECTORY.SizeOfData 
-                writeInt32 os2 debugDataChunk.addr  // IMAGE_DEBUG_DIRECTORY.AddressOfRawData 
-                writeInt32 os2 (textV2P debugDataChunk.addr)// IMAGE_DEBUG_DIRECTORY.PointerToRawData 
-
-                (* dprintf "idd.iddCharacteristics = %ld\n" idd.iddCharacteristics
-                dprintf "iddMajorVersion = %ld\n" idd.iddMajorVersion
-                dprintf "iddMinorVersion = %ld\n" idd.iddMinorVersion
-                dprintf "iddType = %ld\n" idd.iddType
-                dprintf "iddData = (%A) = %s\n" idd.iddData (System.Text.Encoding.UTF8.GetString idd.iddData) *)
+            // write the IMAGE_DEBUG_DIRECTORY 
```

Comment:
Please minimize this diff by replacing the try-with with a bing-end block and keeping the indentation the same